### ISSUE TITLE
mvn package not generating compiled annotations

### ISSUE THREAD

**Issue by sandiindia** [thomasjallerton/nimbus-core#1]
I am getting following error when I run the deploy

[error] java.nio.file.NoSuchFileException: target\generated-sources\annotations\nimbus\nimbus-state.json
	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
	at sun.nio.fs.WindowsFileSystemProvider.newByteChannel(WindowsFileSystemProvider.java:230)
	at java.nio.file.Files.newByteChannel(Files.java:361)
	
....

It seems mvn package is not generating target/generated-sources/annotations/

Here is my pom.xml

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.sandiindia</groupId>
  <artifactId>serverlesstest</artifactId>
  <version>1.0</version>
 
 <dependencies>
 <dependency>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-core</artifactId>
    <version>0.6</version>
</dependency>
</dependencies> 

<build>
<plugins>
<plugin>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-deployment-maven-plugin</artifactId>
    <version>0.3</version>
    <configuration>
        <region>us-east-1</region>
        <stage>dev</stage>
    </configuration>
</plugin>

<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-shade-plugin</artifactId>
    <version>3.2.1</version>
    <executions>
        <execution>
            <phase>package</phase>
            <goals>
                <goal>shade</goal>
            </goals>
            <configuration>
                <outputFile>target/functions.jar</outputFile>
            </configuration>
        </execution>
    </executions>
</plugin>

</plugins>
</build>
</project>

Can anyone please help?

**Comment by sandiindia** [comment]
Here is my pom.xml

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.sandiindia</groupId>
  <artifactId>serverlesstest</artifactId>
  <version>1.0</version>
 
 <dependencies>
 <dependency>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-core</artifactId>
    <version>0.6</version>
</dependency>
</dependencies> 

<build>
<plugins>
<plugin>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-deployment-maven-plugin</artifactId>
    <version>0.3</version>
    <configuration>
        <region>us-east-1</region>
        <stage>dev</stage>
    </configuration>
</plugin>

<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-shade-plugin</artifactId>
    <version>3.2.1</version>
    <executions>
        <execution>
            <phase>package</phase>
            <goals>
                <goal>shade</goal>
            </goals>
            <configuration>
                <outputFile>target/functions.jar</outputFile>
            </configuration>
        </execution>
    </executions>
</plugin>

</plugins>
</build>
</project>

**Comment by sandiindia** [comment]
In my case source is generated in target/classes folder so I guess documentation needs to be updated

**Comment by thomasjallerton** [comment]
Nimbus writes the nimbus folder while doing annotation processing. The path of the folder depends on the compiler, and for Java is normally "target/generated-sources/annotations". I know for Kotlin this changes to "target/generated-sources/kapt/annotations". If you look through the target directory, or wherever the code compiles to, can you find the nimbus directory? It should have some cloudformation files inside of it. If you can find it, and the cloudformation and nimbus-state files are present, you should be able to use the `compiledSourceDirectory` configuration property of the deployment plugin to specify the location of the nimbus folder. E.g. when using kotlin:
```
<plugin>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-deployment-maven-plugin</artifactId>
    <version>0.3</version>
    <configuration>
        <region>eu-west-1</region>
        <compiledSourceDirectory>
            target/generated-sources/kapt/compile/
        </compiledSourceDirectory>
    </configuration>
</plugin>
```

If you can't find the directory it may mean that the annotation processor isn't running for some reason. What are you trying to deploy? i.e. what resources are you trying to deploy, with what annotations? You can also try specifying the exact annotation processors to use in maven (though it should have been detected automatically). 

Finally what OS are you using, in case there is some issue finding the file with the given path.

**Comment by thomasjallerton** [comment]
Ah just seen your update. In this case then just set using the `compiledSourceDirectory` parameter as mentioned above. For example: 
```<plugin>
    <groupId>com.nimbusframework</groupId>
    <artifactId>nimbus-deployment-maven-plugin</artifactId>
    <version>0.3</version>
    <configuration>
        <region>eu-west-1</region>
        <compiledSourceDirectory>
            target/classes/
        </compiledSourceDirectory>
    </configuration>
</plugin>
```

**Comment by sandiindia** [comment]
I changed **compiledSourceDirectory** as you suggested and its working for me now. Thanks! just a little observation that trailing '/' (target/classes**/**) is a must otherwise it will throw that same error

**Comment by thomasjallerton** [comment]
Ah good catch! Will fix that soon. Glad I could help!

**Comment by thomasjallerton** [comment]
Fixed necessary trailing "/" in most recent commit of nimbusdeployment (https://github.com/thomasjallerton/nimbusdeployment/commit/a71224ec06d515550d4d33e5188917c1b01fe787)
Will be included in next release (0.4)
